test/unit/error.spec.ts

```diff
@@ -292,3 +285,88 @@ test("return from nested finally", () => {
     `;
     expect(util.transpileAndExecute(code)).toBe("finally AB");
 });
+
+test("throw and catch custom error object", () => {
+    util.testFunction`
+        try {
+            throw { x: "Hello error object!" };
+        } catch (error) {
+            return error.x;
+        }
+    `.expectToMatchJsResult();
+});
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as classes",
+    errorType => {
+        util.testFunction`
+            try {
+                throw new ${errorType}("message")
+            } catch (error) {
+                if (error instanceof Error) {
+                    return \`\${error}\`;
+                } else {
+                    throw Error("This error serves to prevent false positives from .expectNoExecutionError()");
+                }
+            }
+        `.expectToMatchJsResult();
+    }
+);
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as functions",
+    errorType => {
+        util.testFunction`
+            try {
+                throw ${errorType}("message")
+            } catch (error) {
+                if (error instanceof Error) {
+                    return \`\${error}\`;
+                } else {
+                    throw Error("This error serves to prevent false positives from .expectNoExecutionError()");
+                }
+            }
+        `.expectToMatchJsResult();
+    }
+);
+
+test("subclass Error", () => {
+    util.testFunction`
+        class MyError extends Error {
+            name: "MyError"
+        }
+        
+        try {
+            throw new MyError();
+        } catch (error) {
+            if (error instanceof Error) {
+                return \'\$error\';
```

Comment:
Should it be `${error}` like in other cases?

Reply:
Yep, good catch!